tests/Database/DatabaseEloquentModelTest.php

```diff
@@ -1152,6 +1152,40 @@ public function testModelAttributesAreCastedWhenPresentInCastsArray()
 	}
 
 
+	public function testModelAttributeCastingPreservesNull()
```

Comment:
That's not how tests are meant to be though. You're not meant to squash the test suite to try to match the implementation. You're writing tests that check different things. They shouldn't be organised in a way that nessessaraly reflects the implementation.